graph/service.go

```diff
@@ -29,8 +30,22 @@ func (s *TagStore) Lookup(name string) (*types.ImageInspect, error) {
 		return nil, fmt.Errorf("No such image: %s", name)
 	}
 
+	var tags = make([]string, 0)
+
+	s.Lock()
+	for repoName, repository := range s.Repositories {
+		for ref, id := range repository {
+			if id == image.ID {
+				imgRef := utils.ImageReference(repoName, ref)
+				tags = append(tags, imgRef)
+			}
+		}
+	}
+	s.Unlock()
```

Comment:
Then we have to include `sync.RWMutex` to `TagStore` structure instead of `sync.Mutex`. I could do it. Can this change break something?